homeassistant/components/shelly/sensor.py

```diff
@@ -396,6 +438,24 @@ def native_value(self) -> StateType:
 class BlockSleepingSensor(ShellySleepingBlockAttributeEntity, SensorEntity):
     """Represent a block sleeping sensor."""
 
+    entity_description: BlockSensorDescription
+
+    def __init__(
+        self,
+        wrapper: BlockDeviceWrapper,
+        block: Block | None,
+        attribute: str,
+        description: BlockSensorDescription,
+        entry: RegistryEntry | None = None,
+        sensors: Mapping[tuple[str, str], BlockSensorDescription] | None = None,
+    ) -> None:
+        """Initialize the sleeping sensor."""
+        super().__init__(wrapper, block, attribute, description, entry, sensors)
+
+        self._attr_native_unit_of_measurement = description.native_unit_of_measurement
+        if block and (unit_fn := description.unit_fn):
+            self._attr_native_unit_of_measurement = unit_fn(block.info(attribute))
```

Comment:
Moved out of `ShellySleepingBlockAttributeEntity` as `unit_of_measurement` doesn't apply outside sensor platform